### PR TITLE
refactor: rename osmoutils/accum variables and protos for readability and add docs

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

Note that this PR is state-breaking due to proto renames.

However, this PR does not change any execution flow logic other than renaming and adding documentation.

The most common point of confusion for reviewers has been understanding the "custom accumulator" abstraction which is actually needed for signifying growth insider an interval and not full range of rewards.

Another source of uncertainty has been stemming from where we use a "per share" value vs a "total" value.

Therefore, the relevant updates were made and godocs / comments added where applicable.

Updated `go.mod`s for all packages relying on `osmoutils/accum`

Drive by change:
- remove unused `validateAccumulatorValue`

## Testing and Verifying

This change is a trivial rework / code cleanup without any test coverage.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not applicable

### CODE REVIEW

File: x/concentrated-liquidity/fees.go [hunk 1]
```diff
@@ -309,8 +309,15 @@ func preparePositionAccumulator(accumulator accum.AccumulatorObject, positionKey
 		return err
 	}
 
-	customAccumulatorValue := position.InitAccumValue.Add(growthOutside...)
-	err = accumulator.SetPositionCustomAcc(positionKey, customAccumulatorValue)
+	// The reason for adding the growth outside to the position's initial accumulator value is as follows:
+	// - At any time in-between position updates or claiming, a position must have its InitAccumValue be equal to growth_inside_at_{last time of update}.
```
Comment (p0mvn) [inline x/concentrated-liquidity/fees.go:313]:
```suggestion
	// - At any time in-between position updates or claiming, a position must have its AccumValuePerShare be equal to growth_inside_at_{last time of update}.
```

File: x/concentrated-liquidity/fees.go [hunk 2]
```diff
@@ -309,8 +309,15 @@ func preparePositionAccumulator(accumulator accum.AccumulatorObject, positionKey
 		return err
 	}
 
-	customAccumulatorValue := position.InitAccumValue.Add(growthOutside...)
-	err = accumulator.SetPositionCustomAcc(positionKey, customAccumulatorValue)
+	// The reason for adding the growth outside to the position's initial accumulator value is as follows:
```
Comment (p0mvn) [inline x/concentrated-liquidity/fees.go:312]:
```suggestion
	// The reason for adding the growth outside to the position's initial accumulator value per share is as follows:
```

File: x/concentrated-liquidity/fees.go [hunk 3]
```diff
@@ -309,8 +309,15 @@ func preparePositionAccumulator(accumulator accum.AccumulatorObject, positionKey
...
+	// - At any time in-between position updates or claiming, a position must have its InitAccumValue be equal to growth_inside_at_{last time of update}.
+	// - Prior to claiming (the logic below) updates the position's accumulator to:
```
Comment (p0mvn) [inline x/concentrated-liquidity/fees.go:314]:
```suggestion
	// - Prior to claiming (the logic below), the position's accumulator is updated to:
```

File: osmoutils/accum/errors.go [hunk 4]
```diff
@@ -19,12 +19,12 @@ func (e NoPositionError) Error() string {
...
-type NegativeCustomAccError struct {
-	CustomAccumulatorValue sdk.DecCoins
+type NegativeIntervalAccumulationPerShareError struct {
+	IntervalAccumulationPerShare sdk.DecCoins
 }
 
-func (e NegativeCustomAccError) Error() string {
-	return fmt.Sprintf("customAccumulatorValue must be non-negative, was (%s)", e.CustomAccumulatorValue)
+func (e NegativeIntervalAccumulationPerShareError) Error() string {
+	return fmt.Sprintf("interval accumulaton per share must be non-negative, was (%s)", e.IntervalAccumulationPerShare)
```
Comment (p0mvn) [inline osmoutils/accum/errors.go:27]:
```suggestion
	return fmt.Sprintf("interval accumulation per share must be non-negative, was (%s)", e.IntervalAccumulationPerShare)
```

File: osmoutils/accum/accum_helpers.go [hunk 5]
```diff
@@ -38,26 +38,11 @@ func GetPosition(accum AccumulatorObject, name string) (Record, error) {
...
 	totalRewards = totalRewards.Add(accumulatorRewards...)
 
 	return totalRewards
 }
-
-// validateAccumulatorValue validates the provided accumulator.
-// All coins in custom accumulator value must be non-negative.
-// Custom accumulator value must be a superset of the old accumulator value.
-// Fails if any coin is negative. On success, returns nil.
-func validateAccumulatorValue(customAccumulatorValue, oldPositionAccumulatorValue sdk.DecCoins) error {
```
Comment (p0mvn) [inline osmoutils/accum/accum_helpers.go:54]: Note: this was unused. I remember adding it but it ended up being incorrect to have this validation

**Review by czarcas7ic — APPROVED** [review]
These do feel like a better naming convention and the comments help out as well 👍

File: osmoutils/accum/accum.go [hunk 6]
```diff
@@ -117,33 +117,35 @@ func (accum *AccumulatorObject) AddToAccumulator(amt sdk.DecCoins) {
...
-	return accum.NewPositionCustomAcc(name, numShareUnits, accum.value, options)
+	return accum.NewPositionIntervalAccumulation(name, numShareUnits, accum.valuePerShare, options)
 }
 
-// NewPositionCustomAcc creates a new position for the given name, with the given number of share units.
+// NewPositionIntervalAccumulation creates a new position for the given name, with the given number of share units.
 // The name can be an owner's address, or any other unique identifier for a position.
-// It sets the position's accumulator to the given value of customAccumulatorValue.
-// All custom accumulator values must be non-negative.
+// It sets the position's accumulator to the given value of intervalAccumulationPerShare.
```
[the remark this answers is not in the document]
Comment (p0mvn) [reply osmoutils/accum/accum.go:125]: Going to proceed with merging to avoid conflicts but please lmk, and I can update in a follow-up

**Pull request merged**